### PR TITLE
Set up GAE deployment

### PR DESCRIPTION
**Pull request by nolanbconaway**
Currently deployed manually at: https://sqlfluff-online.uc.r.appspot.com

Uses free-tier (i think?) F1 instance.

Remaining config:

1. set up custom domain, maybe connected with sqlfluff.com?
2. grant admin to maintainers here, or place the gcloud project within the fluff org somehow
3. update refs in docs
4. confirm that free tier is truly being used!

### CODE REVIEW

File: app.yaml
Comment (NiallRees) [inline app.yaml:4]:
```suggestion
automatic_scaling:
  max_instances: 1
```

Comment (NiallRees) [reply app.yaml:4]: The free tier gives 28 instance hours a day, but by default it will scale based on load. This ensures that no matter what happens this app won't go beyond 1 instance

Comment (nolanbconaway) [reply app.yaml:4]: Good call! I am also not able to become 100% certain that I _am_ on the free tier. I mightve paid GCP for something or other a long time ago and am no longer eligible. @NiallRees, you seem to know a little more about GCP than I, do you know of a way to check?

If I am not eligible, does anyone else in @sqlfluff/sqlfluff-maintainers have a fresh GCP account? Alternatively, we can:

1. Try pythonanywhere or heliohosting
2. I've implemented my second most traffic heavy app locally, so i expect my heroku hours to drop a lot; which means there's room for sqfluff online.

Comment (NiallRees) [reply app.yaml]: Everyone gets a free 28 hours of F1 a day from the looks of things. So we should be cool. But wise to set an alert on spend > $0 so that you immediately find out if something goes weird.

Comment (nolanbconaway) [reply app.yaml]: Perfect, I already have said alert set up!

**Comment by nolanbconaway** [comment]
Looks like this checks out! @alanmcruickshank, what are you thoughts about serving the app on a subdomain/directory of the docs page? After we set that up, I can go about editing links in the docs, etc.

Also- I am happy to add whoever as an admin on this GCP project, just tag yourself here!

**Comment by nolanbconaway** [comment]
Alternatively! We can purchase a domain for this for ~$12 USD annually. Alan's already shelled out some cash so i'd be happy to front for a a few years

**Comment by alanmcruickshank** [comment]
Very happy to host this at `online.sqlfluff.com`. Just let me know where to point the DNS and I can set it up 👍

**Comment by alanmcruickshank** [comment]
In fact.... I just set it up as a CNAME record: http://online.sqlfluff.com/

I think it's getting SSH errors, what's the best way to allow that domain on whatever SSH is set up on that app?

**Comment by nolanbconaway** [comment]
Nice! I am not super familiar with GCP app setup but am certain this sort of thing would be supported. I'll look later today, but will also set you up as an editor for the app in case you'd like to look prior to EOD eastern time.

**Comment by alanmcruickshank** [comment]
Ah excellent, I saw the invite - I've followed the instructions online and I think it should now be set up to work. It's not working right now but I imagine DNS setting and SSL issues might take a little time to resolve.

**Comment by nolanbconaway** [comment]
Working for me now! This is excellent! Would love a quick walkthrough on what exactly you configured (just for my education). I'll set up PRs changing out link addresses before merging this. Then, i'll set up a redirect on the old heroku site.

**Comment by alanmcruickshank** [comment]
Followed these instructions: https://cloud.google.com/appengine/docs/standard/python/securing-custom-domains-with-ssl

**Comment by NiallRees** [comment]
I wonder how we can enforce https? Are you using cloudflare @alanmcruickshank ?
image

**Comment by nolanbconaway** [comment]
PR ready! https://github.com/sqlfluff/sqlfluff/pull/1123

**Review by NiallRees — APPROVED** [review]

**Comment by nolanbconaway** [comment]
Pulling the trigger! I'll teardown/redirect traffic from heroku starting later tonight

**Pull request merged**
